tests/e2e/conftest.py

```
@@ -0,0 +1,282 @@
+            "certutil -addstore Root failed — this requires administrator privileges.\n"
+            f"stdout: {result.stdout.strip()}\n"
+            f"stderr: {result.stderr.strip()}\n"
+            "Run the test process as Administrator, or import the cert manually:\n"
+            f"  certutil -addstore Root \"{cert_path}\""
+        )
+    return _cert_sha1_thumbprint(cert_path)
+
+
+def _remove_cert_from_root_store(thumbprint: str) -> None:
+    """Remove cert by SHA-1 thumbprint from LocalMachine\\Root; ignore errors."""
+    subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)